src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -55,72 +110,645 @@ static InitChainEntry D_80936754[] = {
+                EffectSsGSplash_Spawn(globalCtx, &contactPos, NULL, NULL, 0, 190);
+            }
+            contactPos.x = this->actor.world.pos.x;
+            contactPos.z = this->actor.world.pos.z;
+            EffectSsGSplash_Spawn(globalCtx, &contactPos, NULL, NULL, 0, 280);
+            EffectSsGRipple_Spawn(globalCtx, &contactPos, 300, 700, 0);
+            this->actor.terminalVelocity = -3.0f;
+            this->actor.velocity.x *= 0.1f;
+            this->actor.velocity.y *= 0.4f;
+            this->actor.velocity.z *= 0.1f;
+            this->actor.gravity *= 0.5f;
+            rotSpeedX >>= 1;
+            rotSpeedXtarget >>= 1;
+            rotSpeedY >>= 1;
+            rotSpeedYtarget >>= 1;
+            this->actor.bgCheckFlags &= ~0x40;
+            SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 40, NA_SE_EV_DIVE_INTO_WATER_L);
+        }
+        EnKusa_UpdateVelY(this);
+        Math_StepToS(&rotSpeedX, rotSpeedXtarget, 0x1F4);
+        Math_StepToS(&rotSpeedY, rotSpeedYtarget, 0xAA);
+        this->actor.shape.rot.x += rotSpeedX;
+        this->actor.shape.rot.y += rotSpeedY;
+        EnKusa_RandScaleVecToZero(&this->actor.velocity, 0.05f);
+        Actor_UpdatePos(&this->actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 7.5f, 35.0f, 0.0f, 0xC5);
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
+
+void EnKusa_SetupCut(EnKusa* this) {
+    switch (GET_KUSA_TYPE(this)) {
+        case ENKUSA_TYPE_GRASS:
+        case ENKUSA_TYPE_GRASS_2:
+            this->actionFunc = EnKusa_DoNothing;
+            break;
+        case ENKUSA_TYPE_REGROW_GRASS:
+            this->actionFunc = EnKusa_CutWaitRegrow;
+            break;
+    }
+    this->timer = 0;
+}
+
+/**
+ * @brief When cut, grass leaves behind a stump that will begin the regrow process after 120 frames
+ *
+ */
+void EnKusa_CutWaitRegrow(EnKusa* this, GlobalContext* globalCtx) {
+    this->timer++;
+    if ((this->timer) >= 120) {
+        EnKusa_SetupRegrow(this);
+    }
+}
+
+void EnKusa_DoNothing(EnKusa* this, GlobalContext* globalCtx) {
+}
+
+/**
+ * @brief Creates a small scale version of grass underground
+ *
+ */
+void EnKusa_SetupUprootedWaitRegrow(EnKusa* this) {
+    this->actor.world.pos.x = this->actor.home.pos.x;
+    this->actor.world.pos.y = this->actor.home.pos.y - 9.0f;
+    this->actor.world.pos.z = this->actor.home.pos.z;
+    EnKusa_SetScaleSmall(this);
+    this->timer = 0;
+    this->actor.shape.rot = this->actor.home.rot;
+    this->actionFunc = EnKusa_UprootedWaitRegrow;
+}
+
+/**
+ * @brief Window of 50 frames before the grass surfaces to then regrow
+ *
+ */
+void EnKusa_UprootedWaitRegrow(EnKusa* this, GlobalContext* globalCtx) {
+    this->timer++;
+    if ((this->timer) > 120) {
```

Comment:
```suggestion
    if (this->timer > 120) {
```

Reply:
Doesn't look to have been addressed. Forgot to commit/push?